src/main/java/io/pinecone/PineconeIndexOperationClient.java

```diff
@@ -1,136 +1,88 @@
 package io.pinecone;
 
-import com.fasterxml.jackson.databind.ObjectMapper;
 import io.pinecone.exceptions.FailedRequestInfo;
 import io.pinecone.exceptions.HttpErrorMapper;
-import io.pinecone.exceptions.PineconeConfigurationException;
-import io.pinecone.model.ConfigureIndexRequest;
-import io.pinecone.model.CreateIndexRequest;
-import io.pinecone.model.IndexMeta;
 import okhttp3.*;
-
-import java.io.IOException;
-import java.util.ArrayList;
-import java.util.List;
+import org.openapitools.client.ApiClient;
+import org.openapitools.client.ApiException;
+import org.openapitools.client.api.ManageIndexesApi;
+import org.openapitools.client.model.ConfigureIndexRequest;
+import org.openapitools.client.model.CreateIndexRequest;
+import org.openapitools.client.model.IndexList;
+import org.openapitools.client.model.IndexModel;
 
 public class PineconeIndexOperationClient {
-    private final OkHttpClient client;
-    private final PineconeClientConfig clientConfig;
-    private final String url;
-    public static final String ACCEPT_HEADER = "accept";
-    public static final String API_KEY_HEADER_NAME = "Api-Key";
-    public static final String BASE_URL_PREFIX = "https://controller.";
-    public static final String BASE_URL_SUFFIX = ".pinecone.io/databases";
-    public static final String CONTENT_TYPE = "content-type";
-    public static final String CONTENT_TYPE_JSON = "application/json";
-    public static final String CONTENT_TYPE_CHARSET_UTF8 = "charset=utf-8";
-    public static final String EMPTY_RESOURCE_PATH = "";
-    public static final String HTTP_METHOD_DELETE = "DELETE";
-    public static final String HTTP_METHOD_GET = "GET";
-    public static final String HTTP_METHOD_PATCH = "HTTP_METHOD_PATCH";
-    public static final String HTTP_METHOD_POST = "POST";
-    public static final String TEXT_PLAIN = "text/plain";
-    public static final String USER_AGENT_HEADER = "User-Agent";
-
-
-    private PineconeIndexOperationClient(PineconeClientConfig clientConfig, OkHttpClient client, String url) {
-        this.client = client;
-        this.clientConfig = clientConfig;
-        this.url = url;
-    }
-
-    public PineconeIndexOperationClient(PineconeClientConfig clientConfig, OkHttpClient client) {
-        this(clientConfig, client, createUrl(clientConfig));
-    }
+    private ManageIndexesApi manageIndexesApi;
 
```

Comment:
Allow users to pass in config as well

Reply:
Is this primarily for flexibility while we iterate on the approach? It seems like all it's used for at the moment is passing around the `apiKey`, but I can see we still need it when performing data plane operations.

I think this is fine for now, but my opinion is we should look at refactoring the `PineconeClient` and `PineconeConnectionConfig` when we're looking at the data plane to fully remove `PineconeClientConfig`.

Reply:
Yes, I'll be refactoring all of the other classes as a part of future PR's. This is mainly to unblock you for collections work.